enclave-runtime/src/test/mocks/shielding_crypto_mock.rs

```diff
@@ -0,0 +1,44 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use derive_more::{Display, From};
+use itp_sgx_crypto::ShieldingCrypto;
+use std::vec::Vec;
+
+/// Crypto key mock
+///
+/// mock implementation that does not encrypt
+/// encrypt/decrypt return the input as is
+#[derive(Clone)]
+pub struct ShieldingCryptoMock;
```

Comment:
Mock for `ShieldingCrypto`, where `encrypt` and `decrypt` are no-ops

Reply:
I think we should not have a no-op here because we might not detect that we handle encrypted data accidentally with the mock. What do you think?

Reply:
The mock is only included in the test feature, so it is not possible for the mock to end up in production accidentally (as long as the feature flags are set correctly). But I did think about this as well.. in case the feature flag is not enough for you, what would you rather have?

Reply:
This is not what I mean. When we use generic code that is abstracted over the `StateCrypto` trait. We don't uncover potential errors in these generic functions because our test does not perform encryption/decryption.

So I'd suggest that the `RsaSeal` returns an actual key in the tests as well. But i's  hard-coded and not read from the file.

Reply:
But that's exactly the point of a mock? If we want to be sure our encrypt/decrypt works, we have to test that specifically in that component, not in unit tests of components that depend on encryption?

Reply:
Yes, but I doubt that we should use a mock here. I also think that a mock should mock return a value as if the real function was used. Making this a no-op, would change the behavior. For instance, logical errors would not be uncovered in tests this way:

```
let data = rsa.encrypt(plaintext).unwrap();

// logical error done by the programmer who thinks he is working with plaintext
let my_object = Decode::decode(&mut data.as_slice()).unwrap();
```

This would pass in the test with the no-op, but will fail in production.

Reply:
Alright, you convinced me - we should be able to catch those kinds of error with the mock indeed. I will put in a default Rsa3072 key in the mock and use that `encrypt`/`decrypt`.